File: 100.Same Tree/recursion.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        def ist(p,q):
            if p and q:
                if p.val == q.val:
                    if ist(p.left,q.left):
                        if ist(p.right,q.right):
                            return True
                        else:
                            return False
                    else:
                        return False
                else:
                    return False
            
            if not p and q:
                return False
            if p and not q:
                return False
            return True
        return ist(p,q)

File: 100.Same Tree/recursion.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

bool isSameTree(struct TreeNode* p, struct TreeNode* q) {
    if (p && q){
        if (p->val != q->val) return false;
        else {
            if (isSameTree(p->left,q->left)){
                if (isSameTree(p->right,q->right)) return true;
                else return false;
            }
            else return false;
        }
    }
    if (p && !q) return false;
    if (!p && q) return false;
    return true;
    
}
